parquet-hadoop/src/main/java/org/apache/parquet/hadoop/rewrite/ParquetRewriter.java

```diff
@@ -431,7 +480,27 @@ private void processBlock(
     if (chunk.isEncrypted()) {
       throw new IOException("Column " + chunk.getPath().toDotString() + " is already encrypted");
     }
-    ColumnDescriptor descriptor = outSchema.getColumns().get(outColumnIdx);
+
+    ColumnChunkMetaData chunkColumnsRenamed = chunk;
+    if (renamedColumns != null && !renamedColumns.isEmpty()) {
+      chunkColumnsRenamed = ColumnChunkMetaData.get(
```

Comment:
If we add a field to ColumnChunkMetaData in the future, the current `ColumnChunkMetaData.get` method gets deprecated and we can easily get out of sync here. I'm not sure if it is a good idea to introduce a copy|clone method to it with only the name change.

Reply:
Not sure if I understood correctly, I added `ColumnChunkMetaData.copy()`, let me know if that works.